integration_tests/valset_stress_test.go

```diff
@@ -0,0 +1,116 @@
+package integration_tests
+
+// package imports
+import (
+	"context"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+	gravity "github.com/peggyjv/gravity-bridge/module/v2/x/gravity/types"
+)
+
+// call stress_test for a range of nonce values
+
+/// Write test_valset_update test to get latest nonce value
+func (s *IntegrationTestSuite) TestValsetStressUpdate() {
+	s.Run("Bring up chain, and test the valset update", func() {
+		orch := s.chain.orchestrators[1]
+		keyring := orch.keyring
+
+		orchAddress, err := s.chain.orchestrators[1].keyInfo.GetAddress()
+		s.Require().NoError(err)
+		clientCtx, err := s.chain.clientContext("tcp://localhost:26657", keyring, "orch", orchAddress)
+		s.Require().NoError(err)
+
+		startingNonce, err := s.getLastValsetNonce(gravityContract)
+		s.Require().NoError(err, "error getting starting nonce")
+
+		bondTokens := sdk.TokensFromConsensusPower(50000, sdk.DefaultPowerReduction)
+		bondCoin := sdk.NewCoin("testgb", bondTokens)
+
+		valAddress, err := s.chain.validators[3].keyInfo.GetAddress()
+		s.Require().NoError(err)
+		val := sdk.ValAddress(valAddress)
+
+		// Delegate about 5% of the total staking power.
+		s.Require().Eventuallyf(func() bool {
+			s.T().Logf("Sending in valset request (starting_eth_valset_nonce %d)", startingNonce)
+
+			s.T().Logf("Delegating %v to %v in order to generate a validator set update", bondCoin, orchAddress)
+
+			delegate := types.NewMsgDelegate(orchAddress, val, bondCoin)
+			response, err := s.chain.sendMsgs(*clientCtx, delegate)
+			if err != nil {
+				s.T().Logf("error: %s", err)
+				return false
+			}
+
+			if response.Code != 0 {
+				if response.Code != 32 {
+					s.T().Log(response)
+				}
+				return false
+			}
+			return true
+		}, 5*time.Minute, 10*time.Second, "Delegate to validator failed will retry")
+
+		// Verify that delegation went through.
+		s.T().Logf("verifying delegation")
+		s.Require().Eventuallyf(func() bool {
+
+			s.Require().NoError(err, "error querying delegator bonded validators")
```

Comment:
why do we assert no error before we actually query the delegation?

Reply:
Indeed this part is quite weird. I did some cleanup